encrypt/status.go

```diff
@@ -0,0 +1,280 @@
+	IPsecDecryptInts string         `json:"ipsec-decrypt-interfaces-type,omitempty"`
+	IPsecMaxSeqNum   string         `json:"ipsec-max-seq-number,omitempty"`
+	IPsecKeysInUse   int            `json:"ipsec-keys-in-use,omitempty"`
+	IPsecErrCount    int            `json:"ipsec-errors-count,omitempty"`
+	XfrmErrors       map[string]int `json:"xfrm-errors,omitempty"`
+}
+
+// GetEncryptStatus gets encryption status from all/specific cilium agent pods.
+func (s *Status) GetEncryptStatus(ctx context.Context) error {
+	ctx, cancelFn := context.WithTimeout(ctx, s.params.WaitDuration)
+	defer cancelFn()
+
+	if err := s.initTargetCiliumPods(ctx); err != nil {
+		return err
+	}
+
+	res, err := s.fetchEncryptStatusConcurrently(ctx)
+	if err != nil {
+		return err
+	}
+
+	return s.writeStatus(res)
+}
+
+func (s *Status) fetchEncryptStatusConcurrently(ctx context.Context) (map[string]nodeStatus, error) {
+	// res contains data returned from cilium pod
+	type res struct {
+		nodeName string
+		status   nodeStatus
+		err      error
+	}
+	resCh := make(chan res)
+	var wg sync.WaitGroup
+
+	// max number of concurrent go routines will be number of cilium agent pods
+	wg.Add(len(s.ciliumPods))
+
+	// concurrently fetch state from each cilium pod
+	for _, pod := range s.ciliumPods {
+		go func(ctx context.Context, pod *corev1.Pod) {
+			defer wg.Done()
+
+			st, err := s.fetchEncryptStatusFromPod(ctx, pod)
+			resCh <- res{
+				nodeName: pod.Spec.NodeName,
+				status:   st,
+				err:      err,
+			}
+		}(ctx, pod)
+	}
+
+	// close resCh when data from all nodes is collected
+	go func() {
+		wg.Wait()
+		close(resCh)
+	}()
+
+	// read from the channel till it is closed.
+	// on error, store error and continue to next node.
+	var err error
+	data := make(map[string]nodeStatus)
+	for r := range resCh {
+		if r.err != nil {
+			err = errors.Join(err, r.err)
+			continue
+		}
+		data[r.nodeName] = r.status
+	}
+	return data, err
+}
+
+func (s *Status) fetchEncryptStatusFromPod(ctx context.Context, pod *corev1.Pod) (nodeStatus, error) {
+	cmd := []string{"cilium", "encrypt", "status"}
+	output, err := s.client.ExecInPod(ctx, pod.Namespace, pod.Name, defaults.AgentContainerName, cmd)
+	if err != nil {
+		return nodeStatus{}, fmt.Errorf("failed to fetch encryption status from %s: %v", pod.Name, err)
+	}
+	res, err := nodeStatusFromString(pod.Spec.NodeName, output.String())
+	if err != nil {
+		return nodeStatus{}, fmt.Errorf("failed to parse encryption status from %s: %v", pod.Name, err)
+	}
+	return res, nil
+}
+
+func nodeStatusFromString(node string, str string) (nodeStatus, error) {
+	res := nodeStatus{
+		NodeName:   node,
+		XfrmErrors: make(map[string]int),
+	}
+	lines := strings.Split(str, "\n")
+	for _, line := range lines {
+		parts := strings.Split(line, ":")
+		if len(parts) < 2 {
+			continue
+		}
+		key, value := strings.TrimSpace(parts[0]), strings.TrimSpace(parts[1])
+		switch key {
+		case "Encryption":
```

Comment:
Personally i would prefer to backport [your PR](https://github.com/cilium/cilium/pull/30167) (btw: thanks for this !! :heart_eyes: )  that is introducing JSON support for `cilium-dbg encrypt status` as far back as possible and needed. This way we can ditch parsing the text output.

IMO this should be possible and justifiable as it's only new functionality.

@pchaigno WDYT?

Reply:
@mhofstetter so, we will wait for the cilium agent PR, right?

Reply:
Yes, that's at least what I would prefer. As Paul mentioned, it might be good if you could bring up the discussion regarding the backport in the cilium/cilium PR (pointing to this PR / discussion that makes use of it)

Reply:
@mhofstetter [JSON output PR](https://github.com/cilium/cilium/pull/30167) was merged.
So, now I should wait for the next Cilium release and upgrade [this dependency](https://github.com/cilium/cilium-cli/blob/main/go.mod#L17), right?

Thank you!

Reply:
@viktor-kurchenko Great! Either this or start without the dependency and duplicate the model `IPsecStatus` on the cilium-cli side.

We have to handle the case where Cilium isn't supporting the yaml output or has a different model (missing or new properties) anyway (e.g. an older / newer version - or in this case until it's backported).

Reply:
Ready, I've added models and JSON output parsing support.
So, CLI will try to go with JSON otherwise will use text parsing function.

Reply:
great, thanks!